### ISSUE TITLE
`bytes/equals` incorrectly handles subarrays

### ISSUE THREAD

**Issue by tjjfvi** [denoland/deno_std#3603]
```ts
import { assert, assertEquals } from "jsr:@std/assert"
import { equals } from "jsr:@std/bytes"

const a = new Uint8Array(1001).subarray(1)
const b = new Uint8Array(1000)

a[0] = 123
b[0] = 123

assertEquals(a, b) // ok
assert(equals(a, b)) // error; false negative
```
```ts
import { assert, assertNotEquals } from "jsr:@std/assert"
import { equals } from "jsr:@std/bytes"

const a = new Uint8Array(1001).subarray(1)
const b = new Uint8Array(1000)

a[999] = 123

assertNotEquals(a, b) // ok
assert(!equals(a, b)) // error; false positive
```

The `equals32Bit` implementation [does not account for subarrays]( https://github.com/denoland/deno_std/blob/main/bytes/equals.ts#L24-L25).

**Comment by kt3k** [comment]
Maybe `equals32Bit` is only usable when both a.byteOffset and b.byteOffset are multiple of 4.

cc @Aplet123

**Comment by Aplet123** [comment]
I think it would take some benchmarking to see if it's better to do the 8 bit approach when byteOffset isn't a multiple of 4 vs constructing a new Uint8Array from the sliced one so there's no longer any byte offset.

**Comment by tjjfvi** [comment]
`equals32Bit` could also be usable when `a.byteOffset % 4 === b.byteOffset % 4` with some small tweaks.

**Comment by iuioiua** [comment]
Performance fixes applied in #4635.
